abci/proposals/validate.go

```diff
@@ -1,12 +1,82 @@
 package proposals
 
 import (
+	"fmt"
+
 	cometabci "github.com/cometbft/cometbft/abci/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
+	cometproto "github.com/cometbft/cometbft/proto/tendermint/types"
+
+	"github.com/skip-mev/slinky/abci/strategies/codec"
+	"github.com/skip-mev/slinky/abci/strategies/currencypair"
 	"github.com/skip-mev/slinky/abci/ve"
 )
 
+// PruneExtendedCommitInfo validates each vote-extension in the extended commit, and removes
+// any vote-extensions that are invalid. Removal will effectively treat the validator's
+// vote as absent.
+func (h *ProposalHandler) PruneExtendedCommitInfo(
+	ctx sdk.Context, extendedCommitInfo cometabci.ExtendedCommitInfo,
+) (cometabci.ExtendedCommitInfo, error) {
+	// branch the context for verifying vote-extensions on verify-vote extension state
+	verifyVECtx, err := h.appState.VerifyVoteExtensionState(ctx)
+	if err != nil {
+		h.logger.Error(
+			"failed to retrieve verify vote extension state",
+			"err", err,
+		)
+	}
+
+	// tally total voting-power of valid votes
+	var (
+		votingPowerInCommit, totalVotingPower uint64
+	)
+
+	// Validate all oracle vote extensions.
+	for i, vote := range extendedCommitInfo.Votes {
+		// sum the total voting power
+		totalVotingPower += uint64(vote.Validator.Power)
+		// ignore non-commit votes
+		if vote.BlockIdFlag != cometproto.BlockIDFlagCommit {
+			continue
+		}
+
+		// validate the vote-extension
+		if err := validateVoteExtension(verifyVECtx, vote, h.voteExtensionCodec, h.currencyPairStrategy); err != nil {
+			h.logger.Error(
+				"failed to validate vote extension",
+				"err", err,
+				"validator", vote.Validator.Address,
+			)
+
+			// failed to validate this vote-extension, mark it as absent in the original commit
+			vote.BlockIdFlag = cometproto.BlockIDFlagAbsent
```

Comment:
Is this the correct behavior? Based on my understanding, this will change how the x/sla module reasons about validators' behavior w/r/t their commitments. 

Does setting absent here mean a validator will be counted as having voted in consensus, but not submitted a VE i.e. a violation of the SLA for that round?

Reply:
No this indicates that the proposer did not receive this validator's vote